### ISSUE TITLE
Getting this weird error in my app after switching channels or running Flutter upgrade.

### ISSUE THREAD

**Issue by zurie** [fluttercommunity/state_persistence#1]
This happened randomly to me once before.. no idea how i Fixed it.. i thought i just did a clean get from GIT and it worked but not sure anymore..   

I have 2 computers / Mac PC.   the PC use to have this issue which i randomly fixed..  then a week or so later, the mac started getting it.

Oddly enough I can take the same code on mac and PC and the Pc currently is working fine in the emulator.  but the mac / my iphone XR simulator / iphone SE simulator (all sims) and the Device iPhone 7+ all exhibit the same behavior (coming from the same computer).

If i make a page that dumps the data, the working side shows tons of data, but on the mac where its broken I get back {} for the data..     any way i can do more debugging or any ideas?

```
16:28:35.760 23 info flutter.tools 
16:28:35.760 24 info flutter.tools  When the exception was thrown, this was the stack:
16:28:35.761 25 info flutter.tools  #0      _ImmutableMap.[]= (dart:core-patch/immutable_map.dart:74:5)
16:28:35.761 26 info flutter.tools  #1      PersistedData.[]= (package:state_persistence/state_persistence.dart:142:10)
16:28:35.761 27 info flutter.tools  #2      _FlashControlPanelState.build.<anonymous closure>.<anonymous closure> (package:qc/flashcards/flashcard_drawer.dart:86:22)
16:28:35.761 28 info flutter.tools  #3      State.setState (package:flutter/src/widgets/framework.dart:1117:30)
16:28:35.761 29 info flutter.tools  #4      _FlashControlPanelState.build.<anonymous closure> (package:qc/flashcards/flashcard_drawer.dart:85:15)
16:28:35.761 30 info flutter.tools  #5      _SliderState._handleChanged (package:flutter/src/material/slider.dart:402:14)
16:28:35.761 31 info flutter.tools  #6      _RenderSlider._startInteraction (package:flutter/src/material/slider.dart:860:7)
16:28:35.761 32 info flutter.tools  #7      _RenderSlider._handleDragStart (package:flutter/src/material/slider.dart:890:54)
16:28:35.761 33 info flutter.tools  #8      DragGestureRecognizer.acceptGesture.<anonymous closure> (package:flutter/src/gestures/monodrag.dart:218:47)
16:28:35.761 34 info flutter.tools  #9      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:166:24)
16:28:35.761 35 info flutter.tools  #10     DragGestureRecognizer.acceptGesture (package:flutter/src/gestures/monodrag.dart:218:9)
16:28:35.761 36 info flutter.tools  #11     _CombiningGestureArenaMember.acceptGesture (package:flutter/src/gestures/team.dart:41:13)
16:28:35.761 37 info flutter.tools  #12     GestureArenaManager._resolveInFavorOf (package:flutter/src/gestures/arena.dart:263:12)
16:28:35.761 38 info flutter.tools  #13     GestureArenaManager._resolve (package:flutter/src/gestures/arena.dart:222:9)
16:28:35.761 39 info flutter.tools  #14     GestureArenaEntry.resolve (package:flutter/src/gestures/arena.dart:52:12)
16:28:35.761 40 info flutter.tools  #15     _CombiningGestureArenaMember._resolve (package:flutter/src/gestures/team.dart:78:14)
16:28:35.761 41 info flutter.tools  #16     _CombiningGestureArenaEntry.resolve (package:flutter/src/gestures/team.dart:16:15)
16:28:35.761 42 info flutter.tools  #17     OneSequenceGestureRecognizer.resolve (package:flutter/src/gestures/recognizer.dart:240:13)
16:28:35.761 43 info flutter.tools  #18     DragGestureRecognizer.handleEvent (package:flutter/src/gestures/monodrag.dart:193:11)
16:28:35.761 44 info flutter.tools  #19     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
16:28:35.761 45 info flutter.tools  #20     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
16:28:35.761 46 info flutter.tools  #21     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:221:19)
16:28:35.761 47 info flutter.tools  #22     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:199:22)
16:28:35.761 48 info flutter.tools  #23     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
16:28:35.761 49 info flutter.tools  #24     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
16:28:35.761 50 info flutter.tools  #25     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
16:28:35.761 51 info flutter.tools  #29     _invoke1 (dart:ui/hooks.dart:233:10)
16:28:35.762 52 info flutter.tools  #30     _dispatchPointerDataPacket (dart:ui/hooks.dart:154:5)
16:28:35.762 53 info flutter.tools  (elided 3 frames from package dart:async)
16:28:35.762 54 info flutter.tools 
16:28:35.762 55 info flutter.tools  Handler: onStart
16:28:35.762 56 info flutter.tools  Recognizer:
16:28:35.762 57 info flutter.tools    HorizontalDragGestureRecognizer#7f4c5(start behavior: start)
```

**Comment by zurie** [comment]
i fixed it by doing this...

`        storage: JsonFileStorage(initialData: {}),
`

originally I just had....

`        storage: JsonFileStorage(),
`

Odd? almost like the data.json didn't exist and we needed to kick start it into gear? and that only happened during a flutter channel swap ? update? package upgrade ? some weird needle in a haystack??

**Comment by slightfoot** [comment]
This bug was fixed as-of version `0.0.4` of the package. Sorry for any problems this caused.
